### DISCUSSION TITLE
Connector to expose two new methods to get the table and column metadata.

### DISCUSSION THREAD

**Discussion by v3g42** [getdozer/dozer#906] [Feature Requests]
Currently we have the following methods to get metadata about `TableInfo` and `ColumnInfo`
```
fn get_tables(&self, tables: Option<&[TableInfo]>) -> Result<Vec<TableInfo>, ConnectorError>;

pub struct TableInfo {
    pub name: String,
    pub table_name: String,
    pub id: u32,
    pub columns: Option<Vec<ColumnInfo>>,
}

pub struct ColumnInfo {
    pub name: String,
    pub data_type: Option<String>,
}
```
### Problems
- This call is very slow and fetches all tables and columns at the same time. 
- Not fully implemented for all connectors yet. 
- Primary Key and Index information is not available
- Mapping between source column types and dozer field types is not exposed

I suggest refactoring this into two new methods. 
```
## Get just the names of the tables and table level metadata.
fn get_tables() -> Vec<TableDefinition>

# Get detailed table metadata and each of the field properties. 
fn get_table_definitions(tables: Vec<TableFilter>) -> HashMap<TableIdentifier, ColumnDefinition>
```

These methods could be used before SQL is written in
```
connector ls
Dozer UI
```

**Comment by karolisg** [comment]
Do you suggest having new type `TableDefinition`?
